completed technical task number 7

numerouno-life opened the pull request:
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.

numerouno-life committed:
completed technical task number 7
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.

numerouno-life committed:
completed technical task number 7
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.

numerouno-life committed:
completed technical task number 7
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.
изменил тест add() в InMemoryHistoryManagerTest, гит его не принимал, хотя тест работал исправно

numerouno-life committed:
completed technical task number 7
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.
изменил тест add() в InMemoryHistoryManagerTest, гит его не принимал, хотя тест работал исправно

numerouno-life committed:
completed technical task number 7
добавил getCurrentId и setCurrentId,плюс удалил добавление сабтаски в tasks мапу.
изменил тест add() в InMemoryHistoryManagerTest, гит его не принимал, хотя тест работал исправно

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+        this.file = file.toPath();
+    }
+
+    public void save() {
```
метод не должен быть публичным, так как является частью реализации и не должен быть виден или доступен для внешних потребителей

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+            }
+            for (Epic epic : getEpics()) {
+                bw.append(toString(epic)).append("\n");
+                for (Subtask subtask : getSubtasks()) {
```
этот цикл for() нужно вынести вне цикла for (Epic epic : getEpics()), сейчас подзадачи запишутся в файл в количестве умноженном на количество эпиков

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+
+    }
+
+    String toString(Task task) {
```
этот метод используется только внутри класса => должен быть private

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+        return sb.toString();
+    }
+
+    public static Task fromString(String value) {
```
аналогично

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+                maxId = Math.max(maxId, taskId);
+
+                if (task instanceof Epic) {
+                    fileBackedTaskManager.createEpic((Epic) task);
```
использовать createEpic() в этом случае не получится, поскольку не получится гарантировать, что id не перетрется. Нужно вычитанные задачи класть непосредственно в хранилища

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,202 @@
+        return fileBackedTaskManager;
+    }
+
+    public static String getHeader() {
```
этот метод используется только внутри класса => должен быть private

irek-valiev commented on src/tracker/manager/InMemoryTaskManager.java:
```diff
@@ -186,5 +185,11 @@ public List<Task> getHistory() {
+        return currentId;
+    }
 
+    public void setCurrentId(int id) {
```
этот метод должен быть доступен только наследникам класса InMemoryTaskManager => быть protected

irek-valiev commented on src/tracker/manager/InMemoryTaskManager.java:
```diff
@@ -186,5 +185,11 @@ public List<Task> getHistory() {
         return historyManager.getHistory();
     }
 
+    public int getCurrentId() {
```
такой метод не нужен

numerouno-life committed:
reworked-TS-number-7
изменил модификаторы доступа FileBackedTaskManager
перенес цикл for() в private void save()
создал методы для добавления задач (addTask addSubtask addEpic) и использовал их в FileBackedTaskManager
удалил public int getCurrentId()
переписал тесты в FileBackedTaskManagerTest

numerouno-life committed:
reworked-TS-number-7
изменил модификаторы доступа FileBackedTaskManager
перенес цикл for() в private void save()
создал методы для добавления задач (addTask addSubtask addEpic) и использовал их в FileBackedTaskManager
удалил public int getCurrentId()
переписал тесты в FileBackedTaskManagerTest

numerouno-life committed:
reworked-TS-number-7
изменил модификаторы доступа FileBackedTaskManager
перенес цикл for() в private void save()
создал методы для добавления задач (addTask addSubtask addEpic) и использовал их в FileBackedTaskManager
удалил public int getCurrentId()
переписал тесты в FileBackedTaskManagerTest

irek-valiev commented on src/tracker/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,206 @@
+                maxId = Math.max(maxId, taskId);
+
+                if (task instanceof Epic) {
+                    fileBackedTaskManager.addEpic((Epic) task);
```
Сейчас вычитанные из файла Эпики потеряли данные о привязанных к ним подзадачах. Чтобы наполнить внутреннее хранилище Эпиков (subtasks) нужно пройтись по всем подзадачам из хранилища subtasks, по значению epicId брать конкретный Эпик из хранилища epics и использовать метод addSubtask()

numerouno-life committed:
reworked-TS-number-7
try-3
создал лист subtaskAddToEpic,добавил в него сабтаски и привязал сабтаски к эпикам в отдельном цикле. Subtask теперь кладётся в новый лист
